### ISSUE TITLE
Connect Client-Side Components to Server Logic

### ISSUE THREAD

**Issue by ChayaCodes** [ChayaCodes/trivia-quiz#12]
Currently, most of the components in the client side only handle the UI and are not connected to the server. This issue aims to integrate the necessary server logic into the existing client-side components.

1. - Analyze Components: Identify which components need to communicate with the server.
2. - Connect API Endpoints: Implement API calls to fetch or send data for each relevant component.

     -     Use fetch or libraries like axios for API communication.

3. Handle Responses: Process and display the data returned by the server in the UI.
4. Error Handling: Ensure proper error handling for failed requests.
5. Update State Management: Update React state or context with server data where needed.

**Comment by chagitA** [comment]
I would like to take this issue. Do you have any preferences regarding the libraries or logic that will be used to connect the server to the client?

**Comment by ChayaCodes** [comment]
No, use whichever library you're most comfortable with for managing the API requests.

**Comment by ChayaCodes** [comment]
@chagitA , How it's going?

**Comment by ChayaCodes** [comment]
@chagitA I haven't heard from you and I don't know if you're still working on it, so for now I'm marking the issue as unassigned. If you're still working on it, let me know.

**Comment by ChayaCodes** [comment]
@chagitA , I added base logic to API request in the client, with include credentials, to automatically get the identify in the server.
for this, I updated the instructions to create API request to the server, in easier way,
for example:
```javascript
import React from 'react';
import api from '../api/axiosSetup';

function ExampleComponent() {
    const fetchData = async () => {
        try {
            const response = await api.get('/some/protected/endpoint');
            console.log(response.data);
        } catch (error) {
            console.error('Error fetching data:', error);
        }
    };

    React.useEffect(() => {
        fetchData();
    }, []);

    return (
        <div>
            <h1>Example Component</h1>
        </div>
    );
}

export default ExampleComponent;
```

**Comment by ChayaCodes** [comment]
@chagitA, How it's going?

**Comment by chagitA** [comment]
@ChayaCodes Hey, I'm sorry but I will have to ask you to assign the task to someone else. I'll take shorter issues that fit my time right now.
